### ISSUE TITLE
Visual Studio 2022 CUDA samples building & running fine, but instant-ngp doesn't build

### ISSUE THREAD

**Issue by kketzal** [NVlabs/instant-ngp#132]
Hi, I'm trying to run instant-ngp app on my laptop with NVIDIA Geforce GTX 860M....yeah, it's old but **CUDA 11.6 toolkit** samples compiling well in **Windows 10, Visual Studio 2022** and running fine (nbody sample,, for example).
When I try to build the instant-ngp app an error is showed (i'm using  Developer Command Prompt):

-------------
_C:\Users\Manolo\instant-ngp>cmake . -B build
-- The CUDA compiler identification is unknown
CMake Error at C:/Program Files/CMake/share/cmake-3.23/Modules/CMakeDetermineCUDACompiler.cmake:653 (message):
  The CMAKE_CUDA_ARCHITECTURES:

    50

  do not all work with this compiler.  Try:

  instead.
Call Stack (most recent call first):
  CMakeLists.txt:23 (PROJECT)

-- Configuring incomplete, errors occurred!
See also "C:/Users/Manolo/instant-ngp/build/CMakeFiles/CMakeOutput.log".
See also "C:/Users/Manolo/instant-ngp/build/CMakeFiles/CMakeError.log"._
-----------

I set environmet variables in CmakeList.txt:
----------------
_set(CMAKE_CUDA_COMPILER "C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.6/bin/nvcc.exe")
set(CMAKE_CUDA_ARCHITECTURES "50")
set(CUDA_ARCHITECTURES "50")
set(TCNN_CUDA_ARCHITECTURES "50")
string(APPEND CMAKE_CUDA_FLAGS "-gencode=arch=compute_50,code=\'sm_50,compute_50\'")_
---------------
Any ideas?? 
I have attached error.log and output.log

Thanks!!
[CMakeError.log](https://github.com/NVlabs/instant-ngp/files/8053359/CMakeError.log)
[CMakeOutput.log](https://github.com/NVlabs/instant-ngp/files/8053360/CMakeOutput.log)

**Comment by Hat-The-Second** [comment]
yeah same issue here

**Comment by kketzal** [comment]
Finally, I can compile this project in Visual Studio 2022....but all samples run out of memory in my GPU (GeForce GTX 860M)....I think my GPU is too old..... :-(

**Comment by Hat-The-Second** [comment]
What did you change in your setup that it worked now?

**Comment by kketzal** [comment]
Nothing, only open the app folder in VS 2022 and VS 2022 start automatically to check the project....an error was displayed about PATH environment variable....I added "cl.exe"  to the system PATH:

C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.30.30705\bin\Hostx64\x64

And the project was built.

**Comment by son1cman** [comment]
Can you make a tutorial how you manage to compile it using vs2022?

**Comment by son1cman** [comment]
Severity	Code	Description	Project	File	Line	Suppression State
Error	C1189	#error:  -- unsupported Microsoft Visual Studio version! Only the versions between 2015 and 2019 (inclusive) are supported!	C:\Brivity\AI\instant-ngp\out\build\x64-Debug\instant-ngp	F:\CUDA\include\crt\host_config.h	158

**Comment by Hat-The-Second** [comment]
yeah so you have to uninstall your visual studio 2022 and install 2019 again. 2022 is not supportet yet.

**Comment by 3a1b2c3** [comment]
Dumb one: Compiling runs out of memory, how can I limit resources i t uses in  Visual studio 2019?
